test/apigateway.test.ts

```diff
@@ -0,0 +1,107 @@
+import * as cdk from "aws-cdk-lib";
+import { Match, Template } from "aws-cdk-lib/assertions";
+import { User } from "aws-cdk-lib/aws-iam";
+import { AtatRestApi, RestApiVpcConfiguration } from "../lib/constructs/apigateway";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+
+describe("ATAT apigateway construct creation", () => {
+  test("Ensure AtatRestApi is created", async () => {
+    // GIVEN
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app, "TestStack");
+    // WHEN
+    const testApigw = new AtatRestApi(stack, "SampleApi");
+    testApigw.restApi.root.addMethod("ANY");
+    // THEN
+    const template = Template.fromStack(stack);
+    template.hasResourceProperties("AWS::ApiGateway::RestApi", {});
+  });
+
+  test("Ensure grantOnRoute method properly generates IAM policy with execute-api:Invoke action", async () => {
+    // GIVEN
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app, "TestStack");
+    // WHEN
+    const testApigw = new AtatRestApi(stack, "SampleApi");
+    testApigw.restApi.root.addMethod("ANY");
+    const testUser = new User(stack, "testUser");
+    testApigw.grantOnRoute(testUser, "*", "/");
+    // THEN
+    const template = Template.fromStack(stack);
+    template.hasResourceProperties(
+      "AWS::IAM::Policy",
+      Match.objectLike({
+        PolicyDocument: {
+          Statement: [
+            {
+              Action: "execute-api:Invoke",
+              Effect: "Allow",
+            },
+          ],
+          Version: "2012-10-17",
+        },
+      })
+    );
+  });
+
+  test("Ensures AtatRestApi with vpcConfig in props properly sets EndpointConfiguration to PRIVATE", async () => {
```

Comment:
```suggestion
  test("Ensures AtatRestApi with vpcConfig in props properly sets EndpointConfiguration to PRIVATE", async () => {
```

We're actually testing a little more than that. It may be more appropriate to break this into _two_ `test()`s. One that checks the `EndpointConfiguration` and one that checks the `Policy`.

Reply:
Good idea!